### PR TITLE
Web Mint Link

### PR DESCRIPTION
**Pull request by jakzaizzat**
### Summary of Changes

- Add mint link input to post composer
- Show mint link to feed, nft detail screen and collection screen
- If there is `mint_page_url` in the params, fill in the mint input

### Demo or Before/After Pics

**Feed**

https://github.com/gallery-so/gallery/assets/4480258/1acb95ac-e6a9-4b61-8da1-102a1955bd67

**Community**
CleanShot 2023-12-12 at 18 39 49@2x

**Nft Detail Screen**

CleanShot 2023-12-12 at 18 40 13@2x

### Edge Cases

1. Overwrite mint url in the post composer that not mint fun & zora
2. The token from omnibus contract shouldn't show mint button
3. Only show mint button on eth, arbi, zora and base
4. Add referrer address to the mint link

### Testing Steps

**Normal scenario**

1. Create any post and you should see a mint link toggle button
2. You should see the mint button in the feed / nft detail screen / collection if there is support link

**Share Gallery**

1. Login as any user, prefer robin
2. Paste this url to your browser - http://localhost:3000/~/compose?tokenId=98&contractAddress=0xE480a895DE49B49e37A8f0a8bD7e07FC9844CDb9&chain=Arbitrum&collection_title=Dots&token_title=Dots+%2398&caption=I+just+minted+Dots+%2398+by+wutsit2u+on+Prohibition&mint_page_url=https://prohibition.art/project/heart-craft
3. It should fill in the url

### Checklist

Please make sure to review and check all of the following:

- [x] I've tested the changes and all tests pass.
- [x] (if web) I've tested the changes on various desktop screen sizes to ensure responsiveness.
- [x] (if mobile) I've tested the changes on both light and dark modes.

### CODE REVIEW

**Comment by kaitoo1** [comment]
I think we can lose the hover effect (background circle) on the switch since it clashes with the input
CleanShot 2023-12-12 at 20 51 26@2x

**Comment by kaitoo1** [comment]
I think we want to add a close button to this modal
CleanShot 2023-12-12 at 20 54 25@2x
CleanShot 2023-12-12 at 20 54 30@2x

**Review by kaitoo1 — COMMENTED** [review]
this looks great!!! comments are about ui things - code looks great

File: apps/web/src/components/MintLinkButton.tsx [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+    tokenRef
+  );
+
+  const { showModal } = useModalActions();
+
+  const tokenContractAddress =
+    token?.definition?.community?.contract?.contractAddress?.address ?? '';
+  const tokenChain = token?.definition?.community?.contract?.contractAddress?.chain ?? '';
+  const { url: mintURL, provider: mintProviderType } = getMintUrlWithReferrer(
+    overwriteURL ?? token?.definition?.community?.contract?.mintURL ?? '',
```
Comment (Robinnnnn) [inline apps/web/src/components/MintLinkButton.tsx:62]: want to make sure to do `overwriteURL ||` :)

File: apps/web/src/components/MintLinkButton.tsx [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+      };
+    } else if (mintProviderType === 'FxHash') {
+      return {
+        buttonText: 'mint on fxhash',
+        icon: <FxHashLogoIcon mode={variant === 'primary' ? 'light' : 'dark'} />,
+      };
+    } else if (mintProviderType === 'Prohibition') {
+      return {
+        buttonText: 'mint on prohibition',
+        icon: <ProhibitionLogoIcon mode={variant === 'primary' ? 'light' : 'dark'} />,
```
Comment (Robinnnnn) [inline apps/web/src/components/MintLinkButton.tsx:88]: we should add `EnsembleIcon` as well, will provide more details on slack

File: apps/web/src/components/MintLinkButton.tsx [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+    return colors.black[800];
+  }, [variant]);
+
+  if (MINT_LINK_DISABLED_CONTRACTS.has(tokenContractAddress)) {
+    return null;
+  }
+
+  if (MINT_LINK_CHAIN_ENABLED.indexOf(tokenChain) < 0) {
+    return null;
+  }
```
Comment (Robinnnnn) [inline apps/web/src/components/MintLinkButton.tsx:116]: hypernit: could we convert `MINT_LINK_CHAIN_ENABLED` into a Set, similar to L110 `MINT_LINK_DISABLED_CONTRACTS`?

File: apps/web/src/components/MintLinkButton.tsx (hunk 1 above)
Comment (jakzaizzat) [reply apps/web/src/components/MintLinkButton.tsx]: oh shit. Updated!

File: apps/web/src/components/MintLinkButton.tsx (hunk 2 above)
Comment (jakzaizzat) [reply apps/web/src/components/MintLinkButton.tsx:89]: Added `Ensemble` to web and mobile!

CleanShot 2023-12-13 at 15 33 55@2x


CleanShot 2023-12-13 at 15 33 29@2x

File: apps/web/src/components/MintLinkButton.tsx (hunk 3 above)
Comment (jakzaizzat) [reply apps/web/src/components/MintLinkButton.tsx]: Updated to use `Set`!

**Comment by jakzaizzat** [comment]
@Robinnnnn @kaitoo1 Add other changes as well. If the user turn off the toggle mint link, we didn't show it in the feed. Tested on web and mobile


https://github.com/gallery-so/gallery/assets/4480258/f39afe26-117c-4ef6-919c-0b1f55d64139

File: apps/web/src/components/MintLinkButton.tsx (hunk 2 above)
Comment (jakzaizzat) [reply apps/web/src/components/MintLinkButton.tsx:89]: CleanShot 2023-12-13 at 15 39 37@2x

**Review by Robinnnnn — APPROVED** [review]

**Pull request merged**